src/arch/xtensa/include/arch/lib/cache.h

```diff
@@ -22,7 +22,26 @@
 
 #define SRAM_UNCACHED_ALIAS	0x20000000
 
+#ifdef CONFIG_IMX
+/*
+ * Every 512M in 4GB space has dedicate cache attribute.
+ * 1: write through
+ * 2: cache bypass
+ * 4: write back
+ * F: invalid access
+ */
+extern uint32_t _memmap_cacheattr_reset;
```

Comment:
Here we assume that this are all the possible values. We need to check Xtensa manuals and see if there are any other cache options.

Reply:
Fix looks good, but pls check the values and this can go in rc2.

Reply:
I have studied the values a bit in the manual, and based on my understanding possible values can be:

0,1,3: Cached, various ways (3 may be unsupported by certain systems, 0 is no-allocate so it may not be useful for generic usage)
4,5: Write back (5 is no-allocate so it may not be useful); may be unsupported according to the manual
2: Bypass
E: Cache isolate, supposedly for testing so it's not of interest to us
F: Invalid access
6-D: Reserved, may not appear currently

In practice 1, 2, 4 and F are the only useful ones so the comment is fine only mentioning them.

Reply:
@paulstelian97 can you send us a link with documentation? Also, where to find this info. I grepped yesterday through my xtensa docs but couldn't fine this :).

Reply:
Yes, @paulstelian97 I also looked for this and couldn't find anything useful.
Can you please share with us the doc (link)? Thanks!

Reply:
https://0x04.net/~mwk/doc/xtensa.pdf Section 4.6.3.3 (table 4-104 on page 155 of the manual/179 of the PDF) contains the info. It's the first result on the Google search "xtensa cacheattr register".